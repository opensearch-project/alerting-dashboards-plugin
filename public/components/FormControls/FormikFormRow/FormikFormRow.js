/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiCompressedFormRow } from '@elastic/eui';

const FormikFormRow = ({ children, form, name, rowProps: { isInvalid, error, ...rest } }) => (
  <EuiCompressedFormRow
    id={`${name}-form-row`}
    isInvalid={typeof isInvalid === 'function' ? isInvalid(name, form) : isInvalid}
    error={typeof error === 'function' ? error(name, form) : error}
    {...rest}
  >
    {children}
  </EuiCompressedFormRow>
);

FormikFormRow.propTypes = {
  name: PropTypes.string.isRequired,
  rowProps: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  children: PropTypes.node.isRequired,
};

export default FormikFormRow;
