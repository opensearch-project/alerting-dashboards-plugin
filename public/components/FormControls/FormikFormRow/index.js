/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import FormikFormRow from './FormikFormRow';

export default FormikFormRow;
