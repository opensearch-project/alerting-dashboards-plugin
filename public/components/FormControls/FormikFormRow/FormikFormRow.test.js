/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';

import FormikFormRow from './FormikFormRow';

describe('FormikFormRow', () => {
  const Child = () => <div id="mine">child</div>;
  test('renders', () => {
    const component = (
      <FormikFormRow name="testing" rowProps={{}} form={{}}>
        <Child />
      </FormikFormRow>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
