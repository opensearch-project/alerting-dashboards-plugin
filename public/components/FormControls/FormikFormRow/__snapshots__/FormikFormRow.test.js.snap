// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikFormRow renders 1`] = `
<div
  class="euiFormRow euiFormRow--compressed"
  id="testing-form-row-row"
>
  <div
    class="euiFormRow__fieldWrapper"
  >
    <div
      id="mine"
    >
      child
    </div>
  </div>
</div>
`;
